enhancements/hypershift/support-disabling-image-registry-via-capability.md

```diff
@@ -0,0 +1,504 @@
+`.spec.disabledCapabilities` to `["ImageRegistry"]` as requested by the user
+1. Hypershift control-plane-operator detects the ImageRegistry capability is
+disabled in the `HostedCluster`, and skips creating resources for the Image
+Registry operator, ensuring the `ClusterVersion` resource reflects this
+1. The guest cluster becomes available and functional without the image registry
+1. The hosted control-plane becomes available and functional without the
+cluster image registry operator running
+1. Image Registry operator and operand resources are not present in guest or
+management clusters
+1. Image Registry infrastructure resources are not created in cloud provider
+(i.e. storage bucket, IAM user, etc)
+1. No pull-secrets are created for any service accounts, regardless of when
+the service account is created (install time or runtime)
+
+### API Extensions
+
+```go
+import "github.com/openshift/api/config/v1"
+
+// +kubebuilder:validation:Enum=ImageRegistry
+type OptionalCapability v1.ClusterVersionCapability
+
+const ImageRegistryCapability OptionalCapability = v1.ClusterVersionCapabilityImageRegistry
+
+type HostedClusterSpec struct {
+    // ... existing fields ...
+
+    // +kubebuilder:validation:XValidation:rule="size(self.spec.disabledCapabilities) <= size(oldSelf.spec.disabledCapabilities)",message="Disabling capabilities on running clusters is not supported, only removing capabilities from disabledCapabilities (re-enabling capabilities) is supported."
+    // +optional
+    DisabledCapabilities []OptionalCapability
+}
+```
+
+With the new field in place, the `cvoBootrapScript` needs to create the
+`ClusterVersion` resource with a specific set of enabled capabilities, minus
+the ones specified in `DisabledCapabilities`.
+This function currently doesn't have access to the `HostedClusterSpec`.
+`cvo.ReconcileDeployment` will need to change to allow it access to the set of
+disabled capabilities specified by the customer.
+
+TODO: all the reconcile functions in `hostedcontrolplane_controller.go` take a
+`HostedControlPlane` object, not a `HostedCluster`. My assumption is that the
+`HosterControlPlane` gets created from the `HostedCluster`. If this is correct,
+where does it happen?
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+Hypershift is the target topology of this enhancement.
+
+When the cluster service consumer creates a hosted cluster with the `ImageRegistry`
+capability disabled, the hosted control plane of the hosted cluster will not
+have the image registry running.
+
+When the cluster service consumer reenables the `ImageRegistry` capability on a
+running cluster, the CVO will provision the resources for the cluster-image-registry-operator,
+resulting in new pods in the management cluster. Additionally, the pull secrets
+controller (currently located in the openshift-controller-manager) will also be
+enabled by Hypershift, causing a pod redeployment.
```

Comment:
I think this is not necessary to be included in this section

Reply:
Do you think there's a better section for this info? Maybe `Implementation Details/Notes/Constraints`? 🤔

Reply:
that's certainly a good fit for Implementation Details